cloudinit/config/cc_ansible.py

```diff
@@ -0,0 +1,115 @@
+"""ansible enables running on first boot either ansible-pull"""
+
+from copy import deepcopy
+from logging import Logger
+from textwrap import dedent
+from typing import Tuple
+
+from cloudinit.cloud import Cloud
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.distros import ALL_DISTROS, Distro
+from cloudinit.settings import PER_INSTANCE
+from cloudinit.subp import ProcessExecutionError, subp, which
+
+meta: MetaSchema = {
+    "id": "cc_ansible",
+    "name": "Ansible",
+    "title": "Configure ansible for instance",
+    "description": dedent(
+        """\
+        This module provides ansible integration.``
+
+        Ansible is often used agentless and in parallel
+        across multiple hosts simultaneously. This
+        doesn't fit the model of cloud-init: a single
+        host configuring itself during boot. Instead,
+        this module installs ansible during boot and
+        then uses ``ansible-pull`` to run the playbook
+        repository at the remote URL.
+        """
+    ),
+    "distros": [ALL_DISTROS],
+    "examples": [
+        dedent(
+            """\
+            #cloud-config
+            ansible:
+              install: true
+              pull:
+                url: "https://github.com/holmanb/vmboot.git"
+                playbook-name: ubuntu.yml
+            """
+        ),
+    ],
+    "frequency": PER_INSTANCE,
+}
+
+__doc__ = get_meta_doc(meta)
+
+
+def handle(name: str, cfg: dict, cloud: Cloud, log: Logger, _):
+    ansible_cfg: dict = cfg.get("ansible", {})
+    if ansible_cfg:
+        install, ansible_config = get_and_validate_config(ansible_cfg)
+        install_ansible(cloud.distro, install)
+        run_ansible_pull(ansible_config, log)
+
+
+def get_and_validate_config(cfg: dict) -> Tuple[bool, dict]:
+    pull: dict = cfg.get("pull", {})
+    install: bool = cfg.get("install", False)
```

Comment:
Since ansible can be installed from pip or devel channels instead of just the distro deb/rpm, do we want to make this an install_source enum that would allow for devel vs pip vs distro package?  The cc_chef module does something similar though I'm not sure the best way to do it.

Reply:
I've always preferred distro-tested config management solutions for my OS configs, but you're right that others may prefer the PyPI package instead. Probably best to not be opinionated about this, +1.

Reply:
Yes I think we need a bit of discussion to determine how to best represent how install_type can best imply desired behavior to support 'packages' (distro-based/default) vs pip 
- rename `install` to `install_type`, which could be a reserved string "packages", or something to represent pip-based install per some of the [ansible upstream installation recommendations](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#selecting-an-ansible-package-and-version-to-install)


Also one thing we might want to account for in this PR is how we intend to handle of BSD as [ansible may require additional setup on BSD nodes](https://docs.ansible.com/ansible/latest/user_guide/intro_bsd.html). This may only apply to control node type deployments and interacting with the managed nodes over SSH, so it might be something we put off until we have ansib;e control node support. 

We can warn if we don't have support for it at the moment, but it looks like there are some BSD-isms that cloud-init should ultimately handle before trying to support

Reply:
> so it might be something we put off until we have ansib;e control node support.

+1 sounds good.

This module should work "as-is" on the BSDs without extra support. I'll add that if required in future ansible-related PRs.